lib/Models/Leaflet.js

```diff
@@ -236,34 +236,85 @@ Leaflet.prototype.getContainer = function() {
 /**
  * Zooms to a specified camera view or extent.
  *
- * @param {CameraView|Rectangle} viewOrExtent The view or extent to which to zoom.
+ * @param {CameraView|Rectangle|DataSource} target The view, extent or DataSource to which to zoom.
  * @param {Number} [flightDurationSeconds=3.0] The length of the flight animation in seconds.  Leaflet ignores the actual value,
  *                                             but will use an animated transition when this value is greater than 0.
  */
-Leaflet.prototype.zoomTo = function(viewOrExtent, flightDurationSeconds) {
-    if (!defined(viewOrExtent)) {
-        throw new DeveloperError('viewOrExtent is required.');
+Leaflet.prototype.zoomTo = function(target, flightDurationSeconds) {
+    if (!defined(target)) {
+        throw new DeveloperError('target is required.');
     }
 
-    var extent;
-    if (viewOrExtent instanceof Rectangle) {
-        extent = viewOrExtent;
-    } else {
-        extent = viewOrExtent.rectangle;
-    }
+    var that = this;
 
-    // Account for a bounding box crossing the date line.
-    if (extent.east < extent.west) {
-        extent = Rectangle.clone(extent);
-        extent.east += CesiumMath.TWO_PI;
-    }
+    return when().then(function() {
+        var extent;
 
-    this.map.flyToBounds(rectangleToLatLngBounds(extent), {
-        animate: flightDurationSeconds > 0.0,
-        duration: flightDurationSeconds
+        // Target is a KML data source
+        if (defined(target.entities)) {
+            return zoomToDataSource(that, target, flightDurationSeconds);
+        }
+
+        if (target instanceof Rectangle) {
+            extent = target;
+        } else {
+            extent = target.rectangle;
+        }
+
+        // Account for a bounding box crossing the date line.
+        if (extent.east < extent.west) {
+            extent = Rectangle.clone(extent);
+            extent.east += CesiumMath.TWO_PI;
+        }
+    
+        that.map.flyToBounds(rectangleToLatLngBounds(extent), {
+            animate: flightDurationSeconds > 0.0,
+            duration: flightDurationSeconds
+        });
     });
+
 };
 
+function toDegrees(radians) {
+    return radians * 180 / Math.PI;
+}
+
+function entitiesToLatLngBounds(entities) {
+    var maxBounds;
+
+    for (var i = 0, len = entities.length; i < len; i++) {
+        var latLngs = [];
+
+        if (defined(entities[i].polygon)) {
```

Comment:
There are other types of graphical elements, other than polygons, that should be supported. You can get a list here:
https://cesiumjs.org/Cesium/Build/Documentation/Entity.html?classFilter=Entity

But I think rather than try to support all those things, it might be wise to delegate this to the `LeafletVisualizer`, in the same way `Cesium.js` delegates it to the `DataSourceDisplay`. The visualizer knows the Leaflet elements it creates for each type of graphical primitive and can get a bounding box for them.